### PR TITLE
Added safeguard to LinkTrait to prevent error

### LINKED ISSUE #145: Exception thrown on lines with spaces after list
Rendering markdown as HTML throws an `ErrorException: Uninitialized string offset: 0` in [LinkTrait `identifyReference()`](https://github.com/cebe/markdown/blob/master/inline/LinkTrait.php#L249).

This happens with the following markdown:

```markdown
- List Item

  
Line to make sure empty lines with spaces are kept
```

The white-space is imporant here, which is:

```
- List Item
<doesn't matter, empty line or some spaces>
<at least two spaces>
Line to make sure empty lines with spaces are kept
```

As soon as the above appears in a document, the error occurs.

### PR DESCRIPTION
**Pull request by czim**
This catches potential empty string index ErrorExceptions

Fixes #145

### CODE REVIEW

**Comment by czim** [comment]
Just a note on the checks 'failing': This is just `HHVM is no longer supported on Ubuntu Precise.`.

File: inline/LinkTrait.php [hunk 1]
```diff
@@ -248,6 +248,9 @@ protected function renderImage($block)
 
 	protected function identifyReference($line)
 	{
+		if ('' == $line) {
```
Comment (cebe) [inline inline/LinkTrait.php]: this should never happen, as empty lines are skipped by the Parser. See https://github.com/cebe/markdown/commit/7e6fce6b656603c1289a45a41bca065ee8cf3058 which seems to fix the issue you have reported.

**Comment by cebe** [comment]
> Just a note on the checks 'failing': This is just `HHVM is no longer supported on Ubuntu Precise.`.

yeah I need to clean that up, thanks for the hint!

**Comment by cebe** [comment]
fixed by 4225204, thanks!

**Pull request closed without merging**